Xamarin.Forms.Platform.iOS/CollectionView/TemplatedCell.cs

```diff
@@ -25,18 +27,27 @@ protected TemplatedCell(CGRect frame) : base(frame)
 
 		internal IVisualElementRenderer VisualElementRenderer { get; private set; }
 
+		public override void ConstrainTo(CGSize constraint)
+		{
+			ConstrainedSize = constraint;
+		}
+
 		public override void ConstrainTo(nfloat constant)
 		{
 			ConstrainedDimension = constant;
+
+			// Reset constrained size in case ItemSizingStrategy changes
+			// and we want to measure each item
+			ConstrainedSize = default(CGSize);
```

Comment:
You could test this functionality in CollectionView Galleries -> Item Size Galleries -> ItemSizing Strategy. 

- Changing `ItemSizingStrategy` to measure all should measure each cell in `PreferredLayoutAttributesFittingAttributes`.
- Changing `ItemSizingStrategy` to measure all and then back to measure first should **not** break measure first.